lib/pbench/server/cache_manager.py

```diff
@@ -1433,3 +1508,127 @@ def delete(self, dataset_id: str):
         del self.tarballs[name]
 
         self._clean_empties(tarball.controller_name)
+
+    def reclaim_cache(self, goal_pct: float = 20.0, goal_bytes: int = 0) -> bool:
+        """Reclaim unused caches to free disk space.
+
+        The cache tree need not be fully discovered at this point; we can
+        reclaim cache even on a partial tree. This is driven by discovery of
+        the cache directory tree, looking for <resource_id> directories that
+        contain an unpacked tarball root rather than only the `lock` and
+        `last_ref` files.
+
+        This is a "best effort" operation. It will free unlocked caches, oldest
+        first, until both the goal % and the absolute goal bytes are available,
+        or until there are no more unlocked cached tarballs.
+
+        Args:
+            goal_pct: goal percent of cache filesystem free
+            goal_bytes: goal in bytes
+
+        Returns:
+            True if both goals are met, or False otherwise
+        """
+
+        @dataclass
+        class Candidate:
+            last_ref: float  # last reference timestamp
+            cache: Path  # Unpacked artifact directory
+
+        def reached_goal():
+            usage = shutil.disk_usage(self.cache_root)
+            return bool(usage.free >= goal)
```

Comment:
Doesn't the `>=` operator produce a boolean result?...in which case, do we need/want the `bool()` coercion?